assets/js/googlesitekit-idea-hub-notice.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Idea Hub Notice (for Block Editor) entrypoint.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getItem, setItem } from './googlesitekit/api/cache';
+
+const editorNoticeKey = ( postID ) => {
+	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
+};
+
+const loadIdeaHubNotices = async ( _global = global ) => {
+	const { wp } = _global;
+	const shownNotices = [];
+
+	const hasNotice = ( postID ) => {
+		const notices = wp.data.select( 'core/notices' ).getNotices();
+
+		if ( notices === undefined ) {
+			return undefined;
+		}
+
+		const noticeKey = editorNoticeKey( postID );
+
+		return notices.some( ( { id } ) => id === noticeKey );
+	};
+
+	const listener = async () => {
```

Comment:
This function is a little bit of a mess. Please, update it to have the following flow:

1. Get postID
    * If postID is falsy, return
1. Get the item from cache
    * If cacheHit is truthy, unsubscribe and return
1. Get the hasNotice result
    * If it returns `false` and `shownNotices` contains `noticeKey`, then call `setItem`, unsubscribe and return
1. Get post meta
    * If idea_text is not empty, show the notice

Reply:
Thanks, that's much better! 👍🏻